src/actions/BringMoveSwap.ts

```diff
@@ -0,0 +1,266 @@
+        destinationStyle: this.graph.editStyles.pendingModification0,
+        destinationLineStyle: this.graph.editStyles.pendingLineModification0,
+      };
+    }
+    return {
+      sourceStyle: this.graph.editStyles.pendingModification0,
+      sourceLineStyle: this.graph.editStyles.pendingLineModification0,
+      destinationStyle: this.graph.editStyles.pendingModification1,
+      destinationLineStyle: this.graph.editStyles.pendingLineModification1,
+    };
+  }
+
+  private async decorateTargets(
+    sources: TypedSelection[],
+    destinations: TypedSelection[]
+  ) {
+    const decorationTypes = this.getDecorationStyles();
+    await Promise.all([
+      displayPendingEditDecorations(
+        sources,
+        decorationTypes.sourceStyle,
+        decorationTypes.sourceLineStyle
+      ),
+      displayPendingEditDecorations(
+        destinations,
+        decorationTypes.destinationStyle,
+        decorationTypes.destinationLineStyle
+      ),
+    ]);
+  }
+
+  private getEdits(
+    sources: TypedSelection[],
+    destinations: TypedSelection[]
+  ): Edit[] {
+    const usedSources: TypedSelection[] = [];
+    return flatten(
+      zip(sources, destinations).map(([source, destination]) => {
+        if (source == null || destination == null) {
+          throw new Error("Targets must have same number of args");
+        }
+
+        // Add destination edit
+        const result = [
+          {
+            editor: destination.selection.editor,
+            range: destination.selection.selection,
+            newText: source.selection.editor.document.getText(
+              source.selection.selection
+            ),
+            targetsIndex: 0,
+            originalSelection: destination,
+          },
+        ];
+
+        // Add source edit for move and swap
+        // Prevent multiple instances of the same expanded source.
+        if (this.type !== "use" && !usedSources.includes(source)) {
+          let newText = ""; // Defaults to delete source/move
+          if (this.type === "swap") {
+            newText = destination.selection.editor.document.getText(
+              destination.selection.selection
+            );
+          }
+
+          usedSources.push(source);
+          result.push({
+            editor: source.selection.editor,
+            range: source.selection.selection,
+            newText,
+            targetsIndex: 1,
+            originalSelection: source,
+          });
+        }
+
+        return result;
+      })
+    );
+  }
+
+  private async performDocumentEdits(edits: Edit[]) {
+    return runForEachEditor(
+      edits,
+      (edit) => edit.editor,
+      async (editor, edits) => {
+        await editor.edit((editBuilder) => {
+          edits.forEach((edit) => {
+            if (edit.range.isEmpty) {
+              editBuilder.insert(edit.range.start, edit.newText);
+            } else {
+              editBuilder.replace(edit.range, edit.newText);
```

Comment:
Could do that but dont know if it changes anything?

Reply:
Not sure bout with `delete`, but I know that eg with `insert`, it behaves slightly from a `replace` with empty range:

> Although the equivalent text edit can be made with replace, insert will produce a different resulting selection (it will get moved).
(from [docs](https://code.visualstudio.com/api/references/vscode-api#TextEditorEdit))

Reply:
Yes I know selected or not. But delete is working fine with replace as is. Gonna think about it.